apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2UDTsResourceImpl.java

```diff
@@ -119,86 +119,88 @@ public Response createType(final String keyspaceName, final String udtAddPayload
-
-    QueryOuterClass.Query query =
-        new QueryBuilder()
-            .drop()
-            .type(keyspaceName, typeName)
-            // 15-Dec-2021, tatu: Why no "ifExists" option? SGv1 had none which
-            //    seems inconsistent; would be good for idempotency
-            // .ifExists()
-            .build();
-    /*QueryOuterClass.Response grpcResponse =*/ bridge.executeQuery(query);
-    return Response.status(Response.Status.NO_CONTENT).build();
-  }
-
-  @Override
-  public Response updateType(final String keyspaceName, final Sgv2UDTUpdateRequest udtUpdate) {
+  public Uni<RestResponse<Object>> updateType(
+      final String keyspaceName, final Sgv2UDTUpdateRequest udtUpdate) {
     requireNonEmptyKeyspace(keyspaceName);
     final String typeName = udtUpdate.getName();
     requireNonEmptyTypename(typeName);
 
     List<Sgv2UDT.UDTField> addFields = udtUpdate.getAddFields();
     List<Sgv2UDTUpdateRequest.FieldRename> renameFields = udtUpdate.getRenameFields();
 
-    if ((addFields == null || addFields.isEmpty())
-        && (renameFields == null || renameFields.isEmpty())) {
-      return restServiceError(
+    final boolean hasAddFields = (addFields != null && !addFields.isEmpty());
+    final boolean hasRenameFields = (renameFields != null && !renameFields.isEmpty());
+
+    if (!hasAddFields && !hasRenameFields) {
+      return apiErrorResponseUni(
           Response.Status.BAD_REQUEST,
           "addFields and/or renameFields is required to update an UDT");
     }
 
-    if (addFields != null && !addFields.isEmpty()) {
-      List<Column> columns = columns2columns(addFields);
-      QueryOuterClass.Query query =
-          new QueryBuilder().alter().type(keyspaceName, typeName).addColumn(columns).build();
-      bridge.executeQuery(query);
-    }
-
-    if (renameFields != null && !renameFields.isEmpty()) {
-      Map<String, String> columnRenames =
+    final QueryOuterClass.Query addQuery =
+        hasAddFields
+            ? new QueryBuilder()
+                .alter()
+                .type(keyspaceName, typeName)
+                .addColumn(columns2columns(addFields))
+                .build()
+            : null;
+
+    if (hasRenameFields) {
+      final Map<String, String> columnRenames =
           renameFields.stream()
               .collect(
                   Collectors.toMap(
                       Sgv2UDTUpdateRequest.FieldRename::getFrom,
                       Sgv2UDTUpdateRequest.FieldRename::getTo));
-
-      QueryOuterClass.Query query =
+      final QueryOuterClass.Query renameQuery =
           new QueryBuilder()
               .alter()
               .type(keyspaceName, typeName)
               .renameColumn(columnRenames)
               .build();
-      bridge.executeQuery(query);
+      if (addQuery != null) {
+        return executeQueryAsync(addQuery)
+            .chain(any -> executeQueryAsync(renameQuery))
+            .map(any -> RestResponse.ok());
```

Comment:
Note: MUST use opened-up form and NOT `RestResponse::ok` -- latter will fail Integration Test for some reason that I do not understand.

Reply:
Ah. Realized why -- `RestResponse:ok` would take in `any` as opposed to 0-arg version not taking any payload. And the problem is that returned query results are not JSON serializable.
So no mystery there.